File: src/main/java/com/example/trainingsystem/models/Trainee.java
package com.example.trainingsystem.models;

import com.example.trainingsystem.models.embedded.CompletedClass_Trainee;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotBlank;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode
@Document(collection = "Trainees")
public class Trainee {
    @Id
    @JsonProperty("_id")
    private String _id;

    @NotBlank
    @Indexed(unique = true)
    @JsonProperty("Username")
    private String Username;

    @NotBlank
    @JsonProperty("Password")
    private String Password;

    @JsonProperty("TraineeName")
    private String TraineeName;

    @JsonProperty("SkillSet")
    private List<String> SkillSet;

    @JsonProperty("ListOfClassesCompleted")
    private List<CompletedClass_Trainee> ListOfClassesCompleted; //Add ClassName + TotalScore

    @JsonProperty("Level")
    private String Level;
}

File: TrainingSystem.NetAPI/MongoDB_CRUD/Models/Trainee.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;
using System;


namespace MongoDB_CRUD.Models
{
    public class Trainee
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("Username")]
        public string Username { get; set; }
        [BsonElement("Password")]
        public string Password { get; set; }
        [BsonElement("TraineeName")]
        public string TraineeName { get; set; }
        [BsonElement("SkillSet")]
        public List<string> SkillSet { get; set; }
        [BsonElement("ListOfClassesCompleted")]
        public List<Object> ListOfClassesCompleted { get; set; }
        [BsonElement("Level")]
        public string Level { get; set; }
        [BsonElement("ListOfClassesLearning")]
        public List<string> ListOfClassesLearning { get; set; }
        [BsonElement("ListOfClassesLiked")]
        public List<string> ListOfClassesLiked { get; set; }



        [BsonElement("_class")]
        public string _class { get; set; }




    }
}
